amundsen_application/static/js/ducks/utilMethods.ts

```diff
@@ -25,3 +26,14 @@ export function filterFromObj(initialObj: object, rejectedKeys: string[]): objec
     return obj;
   }, {});
 }
+
+
+export function log(command: string, target_id: string, target_type: string, label?: string, location?: string, index?: number) {
+  const params = { command, target_id, target_type, label, location, index };
+  ['target_type', 'label', 'location', 'index'].forEach((propName) => {
+    if (params[propName] == null || params[propName] === '') {
+      delete params[propName];
+    }
+  });
+  logAction(params);
+}
```

Comment:
This originally started with just 4 parameters, but now it has 6. I'm on the fence on whether to just accept a "params" object and validate fields on that. The downside is that everywhere this is used the caller will have to build an object like

```
{ 
  command: "click", 
  target_id: "nav::browse", 
  target_type: "link",
  label: "Browse",
  location: "navbar" 
}
```

Reply:
I think that going with an object that has a defined interface will be the cleanest way to go. The parameters create a lot of room for error because we will have to remember to call this method with parameters in the right order.